src/pages/filter/common/attributes/base-functions.store.ts

```diff
@@ -0,0 +1,79 @@
+import { makeAutoObservable } from 'mobx'
+
+import { DefaultProblemGroup } from '@core/enum/default-problem-group-enum'
+import { ModeTypes } from '@core/enum/mode-types-enum'
+import datasetStore from '@store/dataset/dataset'
+import { DtreeStatFuncStore } from '@store/dtree/dtree-stat-func.store'
+import { FilterStatFuncStore } from '@store/filter/filter-stat-func.store'
+import {
+  IFuncPropertyStatus,
+  IInheritanceModeArgs,
+  TFuncCondition,
+  TPropertyStatus,
+  TVariant,
+} from '@service-providers/common'
+import { getCurrentModeType } from '@utils/getCurrentModeType'
+import { BaseAttributeStore } from './base-attribute.store'
+
+export class BaseFunctionsStore {
+  readonly attributeStore: BaseAttributeStore
+  public statFuncStore: DtreeStatFuncStore | FilterStatFuncStore
+
+  constructor(
+    attributeStore: BaseAttributeStore,
+    statFuncStore: DtreeStatFuncStore | FilterStatFuncStore,
+  ) {
+    this.attributeStore = attributeStore
+    this.statFuncStore = statFuncStore
+    makeAutoObservable(this)
+  }
+
+  public get attributeStatus(): TPropertyStatus | undefined {
+    return this.attributeStore.attributeStatus
+  }
+
+  public get initialCondition(): TFuncCondition | undefined {
+    return this.attributeStore.initialCondition as TFuncCondition
+  }
+
+  public get attributeName(): string | undefined {
+    return this.attributeStatus?.name
+  }
+
+  public get attributeSubKind(): string | undefined {
+    if (this.attributeStatus && 'sub-kind' in this.attributeStatus) {
+      return this.attributeStatus['sub-kind']
+    }
+
+    return undefined
+  }
+
+  public get problemGroups(): string[] {
+    const status = this.attributeStatus as IFuncPropertyStatus
+    return status?.family || []
+  }
+
+  public get variants(): TVariant[] {
```

Comment:
`variants` getter can be moved to BaseStatFuncStore.
And then it will be possible to remove `statFuncStore` as dependency